docs/build/build_system.md

```diff
@@ -0,0 +1,209 @@
+
+If you wish to use CLI instead, you can list all the available configurations in the `CMakePresets.json` file and pick one of the presets:
+
+```sh
+cmake --list-presets
+```
+
+For example, this is how to build sc-machine in debug mode using dependencies from Conan:
+
+```sh
+# use pipx to install Conan if not already installed
+# Install pipx first using this guide: https://pipx.pypa.io/stable/installation/ 
+pipx install conan
+# Use preset with Conan-provided dependencies and debug build type
+cmake --preset debug-conan
+# Use debug build preset since we've used a debug configure preset
+cmake --build --preset debug
+```
+
+This is how you would build sc-machine in release mode using system-provided dependencies:
+
+```sh
+# release build type with tests (no conan postfix = system-provided deps)
+cmake --preset release-with-tests
+cmake --build --preset release
+```
+
+The build artifacts will be located in the `build/<Release|Debug>/bin` folder.
+
+Default installation folder location (used for package consumption) is `build/<Release|Debug>/install`, installation process can be fired by appending `--target install` to the `cmake --build --preset <preset>` command.
+
+## Consuming the package
+
+<details markdown>
+
+<summary>find_package() using CPack artifacts on GitHub Actions</summary>
+
+The package artifacts are available [here](https://github.com/ostis-ai/sc-machine/releases). To use these in your CMake project, download and extract the release for your platform to any convenient location. Then make it available to CMake by appending folder path to `CMAKE_PREFIX_PATH`:
+
+```cmake
+# import sc-machine from the Releases archive
+# you can override this variable via -D<proj_name>_PATH_SC_MACHINE_PATH or CMakeUserPreset.json / CMakePreset.json files
+set(<proj_name>_SC_MACHINE_PATH "/location/to/sc-machine-<version>-<platform>" CACHE PATH "sc-machine installation path")
+
+# can be overriden using env variables as well
+if(DEFINED ENV{<proj_name>_SC_MACHINE_PATH})
+  set(<proj_name>_SC_MACHINE_PATH "$ENV{<proj_name>_SC_MACHINE_PATH}")
+endif()
+
+list(APPEND CMAKE_PREFIX_PATH ${<proj_name>_SC_MACHINE_PATH})
+```
+</details>
+
+<details markdown>
+
+<summary>find_package() using Conan</summary>
+
+- You want to use a released version of sc-machine
+  
+  Simply add sc-machine as a dependency to your project and use Conan to install it for you.
+
+  conanfile.txt:
+
+  ```ini
+  [requires]
+  sc-machine/0.10.0
+
+  [generators]
+  CMakeDeps
+  CMakeToolchain
+
+  [layout]
+  cmake_layout
+  ```
+
+  ```sh
+  conan install .
+  cmake --preset conan-release
+  cmake --build --preset conan-release
+  ```
+
+- You want to use a package version that is not published in a Conan repo:
+  
+  Export the version you'd like to use to local conan cache:
+  ```sh
+  git clone https://github.com/ostis-ai/sc-machine
+  git checkout <your-commit>
+  cmake --preset release-conan
+  cmake --build --preset release --target install
+  conan export-pkg .
```

Comment:
is this error expected when executing `conan export-pkg .`?
![image](https://github.com/user-attachments/assets/fef0eb0c-6edc-4684-bc18-2f9cea21a13b)

Reply:
if I need to run this command instead of one mentioned in docs then docs should be updated, shouldn't it?

Reply:
You shouldn't run `conan install`, since this step should be handled by `conan_provider.cmake`. While @NikitaZotov's answer is technically not wrong, gtest will build happily with either cppstd, it's just that export-pkg doesn't allow building the packages from sources by default. This case is curious though: I would suppose that running `cmake --preset release-conan` would install all the deps, because it runs `conan install . --build=missing` under the hood, which is why I didn't encounter this error.

Does this replicate on a clean installation if you pick `debug-conan` as the CMake preset?

Reply:
@FallenChromium, configure sc-machine both with `debug-conan` and with `release-conan` presets don't solve this problem. `conan export-pkg .` ends with the same error.  `conan graph explain .` shows this: 

![Снимок экрана от 2024-11-24 13-42-24](https://github.com/user-attachments/assets/90ab1e87-55aa-4ea1-8744-c0c785039d2e)

Reply:
It can be solved if we will change `self.test_requires("gtest/1.14.0")` to `self.build_requires("gtest/1.14.0")` in build_requirements of conanfile.py

Reply:
@kilativ-dotcom, you need to manually set the version in conanfile.py